src/ray/raylet/node_manager.cc

```diff
@@ -2861,9 +2861,8 @@ void NodeManager::HandlePinObjectIDs(const rpc::PinObjectIDsRequest &request,
       if (plasma_results[i].data == nullptr) {
         objects.push_back(nullptr);
       } else {
-        objects.emplace_back(std::unique_ptr<RayObject>(new RayObject(
-            std::make_shared<PlasmaBuffer>(plasma_results[i].data),
-            std::make_shared<PlasmaBuffer>(plasma_results[i].metadata), {})));
+        objects.emplace_back(std::unique_ptr<RayObject>(
+            new RayObject(plasma_results[i].data, plasma_results[i].metadata, {})));
```

Comment:
Just curious, but why do we delete a shared pointer here? Is `plasma_results[i].data` already a pointer (it was hard to know from this PR changes)?

Reply:
yes, the original code just wrap a shared pointer with another shared pointer without actual functionality.